aws_lambda_powertools/utilities/data_classes/api_gateway_authorizer_event.py

```diff
@@ -0,0 +1,483 @@
+
+        method = {"resourceArn": resource_arn, "conditions": conditions}
+        if effect.lower() == "allow":
+            self._allow_methods.append(method)
+        else:  # deny
+            self._deny_methods.append(method)
+
+    @staticmethod
+    def _get_empty_statement(effect: str) -> Dict[str, Any]:
+        """Returns an empty statement object prepopulated with the correct action and the desired effect."""
+        return {"Action": "execute-api:Invoke", "Effect": effect.capitalize(), "Resource": []}
+
+    def _get_statement_for_effect(self, effect: str, methods: List) -> List:
+        """This function loops over an array of objects containing a resourceArn and
+        conditions statement and generates the array of statements for the policy."""
+        if len(methods) == 0:
+            return []
+
+        statements = []
+
+        statement = self._get_empty_statement(effect)
+        for method in methods:
+            if method["conditions"] is None or len(method["conditions"]) == 0:
+                statement["Resource"].append(method["resourceArn"])
+            else:
+                conditional_statement = self._get_empty_statement(effect)
+                conditional_statement["Resource"].append(method["resourceArn"])
+                conditional_statement["Condition"] = method["conditions"]
+                statements.append(conditional_statement)
+
+        if len(statement["Resource"]) > 0:
+            statements.append(statement)
+
+        return statements
+
+    def allow_all_methods(self):
+        """Adds a '*' allow to the policy to authorize access to all methods of an API"""
+        self._add_method("Allow", HttpVerb.ALL, "*", [])
+
+    def deny_all_methods(self):
+        """Adds a '*' allow to the policy to deny access to all methods of an API"""
+        self._add_method("Deny", HttpVerb.ALL, "*", [])
+
+    def allow_method(self, http_method: str, resource: str, conditions: Optional[List[Dict]] = None):
+        """Adds an API Gateway method (Http verb + Resource path) to the list of allowed
+        methods for the policy.
+
+        Optionally includes a condition for the policy statement. More on AWS policy
+        conditions here: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements.html#Condition"""
+        self._add_method("Allow", http_method, resource, conditions or [])
+
+    def deny_method(self, http_method: str, resource: str, conditions: Optional[List[Dict]] = None):
+        """Adds an API Gateway method (Http verb + Resource path) to the list of denied
+        methods for the policy.
+
+        Optionally includes a condition for the policy statement. More on AWS policy
+        conditions here: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements.html#Condition"""
+        self._add_method("Deny", http_method, resource, conditions or [])
```

Comment:
I think `allow_route` is more inline with what we're trying to do here, since `allow_method` is simply a method. Route is composed of a method + path/resource.

Reply:
`allow_all_methods`, and `deny_all_methods`, perhaps need to be `allow_all_routes` and take a method param. If we're looking for convenience we could have a default value for method param as we're doing it here, since that's the most common intent.

Reply:
done.